Add ThingsToDo and Campground Component

rutujapadgilwar opened the pull request:
## **Description**
This pull request adds the functionality for displaying "things to explore" and "campground info" card components. I have used the thingstodo and campground endpoints to fetch the necessary data. 

## **Changes Made** 
1. Added new components "ThingsToExplore" and "Campground and More" that showcases images, short descriptions of various attractions and options for exploration, and a link is provided for users to access more detailed information about each item.
2. Updates to "parkInfoPage" Component - Made improvements and modifications to the existing "parkInfoPage" components to make them media responsive.
3. Addition of "styles.css" File - "styles.css" contains CSS custom color properties.
4. I have removed the LoadCampgroundData file that I previously created, replacing it with the LoadCampground file developed by Shreya. This modification enhances reusability within the project and promotes efficient code management.

## **Screenshots**

<img width="954" alt="image" src="https://github.com/rutujapadgilwar/Park-Dashboard/assets/106363277/0eed7e1c-2069-4ed1-8d18-3e086e485604">
<img width="397" alt="image" src="https://github.com/rutujapadgilwar/Park-Dashboard/assets/106363277/87a712d2-a741-44f0-93e4-6f4abfda6462">

Note - { This pull request builds upon the previous PR commit [commit link: https://github.com/rutujapadgilwar/Park-Dashboard/pull/8/commits/a23c7e9a82c3ee6ddbda7d46ced43dfd271c37d1], incorporating additional updates. Please review the code changes and provide any further feedback or required modifications. }

rutujapadgilwar committed:
Add thingsToDo and Compground Component

schoure98 commented:
Can you add data of things to do in local storage like the previous data we have added? The component is taking more time to load the data.

rutujapadgilwar commented:
> Can you add data of things to do in local storage like the previous data we have added? The component is taking more time to load the data.

I have implemented a similar approach in my code using the loadthingstodoData component. Additionally, I have included an object called thingstodoCatch outside the component, which serves as a cache to store the retrieved thingstodo data.
When the component renders, it first checks if the data is available in the cache. If it is, the component directly sets the data from the cache using sethingsToDoData(thingsToDoDataCatch.data). However, if the data is not found in the cache, the component makes an API call to fetch the data and then stores it in the cache for future use.

It seems that loading the images is causing the delay, rather than the data retrieval itself. I will explore alternative approaches to address this and find ways to optimize the image-loading process for better performance.

schoure98 commented on park-dashboard/src/Components/CampgroundInfo/CampgroundInformation.js:
```diff
@@ -36,7 +40,11 @@ function CampgroundInformation(item) {
           <CardActions>
-            <Button size="small" onClick={() => openUrl(item.url)}>Learn More</Button>
+            {item.regulationsurl && (
+              <Button size="small" onClick={() => openUrl(item.regulationsurl)}>
```
Is it possible to add key to the button? It might be considered as a good practice since it's going to be part of rendering the button multiple times? Attaching the code snippet below, if that helps
`<Button key={item.id} size="small" onClick={() => openUrl(item.regulationsurl)}>
      Learn More
  </Button>`

rutujapadgilwar replied:
Thanks for the suggestion, I have added the key to the button.

schoure98 reviewed (commented):
Overall good work on this component.

rutujapadgilwar committed:
Add noThingsToDo and noCampground message

juhiaugustine commented on park-dashboard/src/Components/CampgroundInfo/CampgroundInformation.js:
Both the Components look neat and nice. One small improvisation would be, If there are no campgrounds available, try adding a message in a card view or something, so that it won't look empty.

![image](https://github.com/rutujapadgilwar/Park-Dashboard/assets/114516950/f192d664-6038-4611-a55c-2441326993ae)

rutujapadgilwar replied:
NoThingsToExplore and NoCampground info found message has been added.

rutujapadgilwar replied:
<img width="1440" alt="image" src="https://github.com/rutujapadgilwar/Park-Dashboard/assets/106363277/f4edc2b7-bb40-4125-8724-d584bb5c513f">

rutujapadgilwar committed:
Add the key to button

rutujapadgilwar committed:
Resolve the conflicts
